src/unwinder/i386/unwinder_i386.cpp

```diff
@@ -72,49 +115,25 @@ OOPStackUnwinderX86::VirtualUnwind(
         *HandlerRoutine = NULL;
     }
 
-    REGDISPLAY rd;
-
-    FillRegDisplay(&rd, ContextRecord);
-
-    rd.SP = ContextRecord->Esp;
-    rd.PCTAddr = (UINT_PTR)&(ContextRecord->Eip);
-
-    if (ContextPointers)
-    {
-        rd.pCurrentContextPointers = ContextPointers;
-    }
-
-    CodeManState codeManState;
-    codeManState.dwIsSet = 0;
-
-    EECodeInfo codeInfo;
-    codeInfo.Init((PCODE) ControlPc);
+    _ASSERTE(ContextRecord->Eip == ControlPc);
 
-    if (!UnwindStackFrame(&rd, &codeInfo, UpdateAllRegs, &codeManState, NULL))
+    if (!OOPStackUnwinderX86::Unwind(ContextRecord, ContextPointers))
     {
         return HRESULT_FROM_WIN32(ERROR_READ_FAULT);
     }
 
-    ContextRecord->ContextFlags |= CONTEXT_UNWOUND_TO_CALL;
-
-#define ARGUMENT_AND_SCRATCH_REGISTER(reg) if (rd.pCurrentContextPointers->reg) ContextRecord->reg = *rd.pCurrentContextPointers->reg;
-    ENUM_ARGUMENT_AND_SCRATCH_REGISTERS();
-#undef ARGUMENT_AND_SCRATCH_REGISTER
-
-#define CALLEE_SAVED_REGISTER(reg) if (rd.pCurrentContextPointers->reg) ContextRecord->reg = *rd.pCurrentContextPointers->reg;
-    ENUM_CALLEE_SAVED_REGISTERS();
-#undef CALLEE_SAVED_REGISTER
-
-    ContextRecord->Esp = rd.SP - codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
-    ContextRecord->Eip = rd.ControlPC;
-
     // For x86, the value of Establisher Frame Pointer is Caller SP
     //
     // (Please refers to CLR ABI for details)
     *EstablisherFrame = ContextRecord->Esp;
     return S_OK;
 }
 
+BOOL DacUnwindStackFrame(T_CONTEXT* pContextRecord, T_KNONVOLATILE_CONTEXT_POINTERS* pContextPointers)
+{
+    return OOPStackUnwinderX86::Unwind(pContextRecord, pContextPointers);
```

Comment:
I am not sure if we want the real context pointers for the DAC unwind. For all other architectures, we just point all the context pointers to point into the pContextRecord. I guess the reason is that if we did it the way you did it here, the context pointers would be target address space addresses while the debugger code expects them to be the debugger address space ones.
@mikem8361 am I right?